ophyd/userapi/scan_api.py

```diff
@@ -229,11 +229,17 @@ def run(self, *args, **kwargs):
         The main loop of the scan. This routine runs the scan and calls the
         ophyd runengine.
         """
+
+        # Raise if no detectors are associated with a scan instance,
+        # and do it prior to context manager __entry__
+        if not len(self.detectors) > 0:
```

Comment:
python nitpick

```python
if not self.detectors:
    raise ...
```

Has the same behaviour.  I think it is slightly more readable, but not a huge issue either way.